src/lib/mapResolvers.js

```diff
@@ -0,0 +1,12 @@
+export default (namespace, resolvers) => {
+  if (resolvers instanceof Object) {
+    for (const type of Object.keys(resolvers)) {
+      for (const field of Object.keys(resolvers[type])) {
+        const fn = resolvers[type][field];
+        resolvers[type][field] = (root, args, context, info) =>
+          fn(root, args, context[namespace], info);
+      }
+    }
+    return resolvers;
+  }
+};
```

Comment:
I have been testing something similar in my environment for a little while and something that I noticed is that this hides errors when the user provides an invalid value for a resolver.

```javascript
const resolvers = {
  TypeName: {
    getName: 'not a resolver',
  },
};
```

Apollo will throw an error during startup; however, wrapping the method hides this issue until the user attempts to execute this handler.

In my case I forgot to extract the parent resolvers:

_MyInterface.js_
```javascript
module.exports = { schema, resolvers };
```

_MyType.js_
```javascript
import MyInterface from './MyInterface';

const schema = gql`
  type MyType implements MyInterface {
    ...
  }
`;

const resolvers = Object.assign({ 
}, MyInterface); // should be MyInterface.resolvers

module.exports = { schema, resolvers };
```

Reply:
@corycook would ensuring that the resolver is a `Function` be enough of a check?

```javascript
export default (namespace, resolvers) => {
  if (resolvers instanceof Object) {
   for (const type of Object.keys(resolvers)) {
     for (const field of Object.keys(resolvers[type])) {
       const fn = resolvers[type][field];
       if (typeof fn !== 'function'){
         throw new Error(`Expected Function for ${type}.${field} resolver but received ${typeof fn}`)
       }
       resolvers[type][field] = (root, args, context, info) =>
         fn(root, args, context[namespace], info);
     }
   }
   return resolvers;
  }
};
```

Reply:
I think this check is sufficient, but you might want to just forward the value to Apollo and let them handle showing the error message in the case that the value is not a function.

Reply:
Then again, throwing the error here might help with unit testing.